shell/platform/android/io/flutter/embedding/engine/dart/DartExecutor.java

```diff
@@ -181,6 +184,22 @@ public void setMessageHandler(@NonNull String channel, @Nullable BinaryMessenger
   }
   //------ END BinaryMessenger -----
 
+  /**
+   * Returns an identifier for this executor's primary isolate.  This identifier can be used
+   * in queries to the Dart service protocol.
+   */
+  public String getIsolateServiceId() {
+    return isolateServiceId;
```

Comment:
This is racy. How is the caller going to know to wait for the isolate launch to be completed?

Reply:
With a getter API the client would need to poll until the isolate ID is available, or otherwise wait for another signal of app liveness such as the first frame callback.  I'll add a callback specifically for the isolate ID.